documentation/guide-repository.asciidoc

```diff
@@ -1,77 +1,119 @@
 * Support for paging is already build-in. Again for static link:guide-jpa-query.asciidoc[query] method the is nothing you have to do except using the paging objects in the signature.
 * Still you have the freedom to write custom implementations via default methods within the repository interface (e.g. for dynamic queries).
 
+== Dependency
+In case you want to switch to or add Spring Data support to your devon4j-spring or Quarkus application, all you need is to add the respective maven dependency:
+
+.**spring**
+[source,xml]
+--------
+<!-- Starter for consuming REST services -->
+<dependency>
+  <groupId>com.devonfw.java.starters</groupId>
+  <artifactId>devon4j-starter-spring-data-jpa</artifactId>
+</dependency>
+--------
+
+.**quarkus**
+[source,xml]
+--------
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-spring-data-jpa</artifactId>
+</dependency>
+--------
+
 == Repository
-For each entity `«Entity»Entity` an interface is created with the name `«Entity»Repository` extending https://github.com/devonfw/devon4j/blob/develop/modules/jpa-spring-data/src/main/java/com/devonfw/module/jpa/dataaccess/api/data/DefaultRepository.java[`DefaultRepository`].
-Such repository is the analogy to a link:guide-dao.asciidoc[Data-Access-Object (DAO)] used in the classic approach or when spring-data is not an option.
+For each entity `«Entity»Entity` an interface is created with the name `«Entity»Repository` extending one of the following Spring Data repositories:
+    * org.springframework.data.repository.Repository
+    * org.springframework.data.repository.CrudRepository
+    * org.springframework.data.repository.PagingAndSortingRepository
+    * org.springframework.data.jpa.repository.JpaRepository
+Such repository is the analogy to a link:guide-dao.asciidoc[Data-Access-Object (DAO)] used in the classic approach or when Spring Data is not an option.
+
+These Spring Data repositories provide some basic implementations for accessing data, e.g. returning all instances of a type (`findAll`) or returning an instance by its ID (`findById`).
+
+In addition, repositories can be enriched with additional functionality, e.g. to add QueryDSL functionality or to override the default implementations, by using so called repository fragments:
 
 === Example
 The following example shows how to write such a repository:
 
+.**Repository**
 [source,java]
 ----
-public interface ExampleRepository extends DefaultRepository<ExampleEntity> {
+public interface ExampleRepository extends JpaRepository<ExampleEntity, Long>, ExampleFragment {
 
-  @Query("SELECT example FROM ExampleEntity example" //
-      + " WHERE example.name = :name")
+  @Query("SELECT example FROM ExampleEntity example WHERE example.name = :name")
   List<ExampleEntity> findByName(@Param("name") String name);
 
-  @Query("SELECT example FROM ExampleEntity example" //
-      + " WHERE example.name = :name")
+  @Query("SELECT example FROM ExampleEntity example WHERE example.name = :name")
```

Comment:
Queries can get very complex and reading them can be hard.
With this formatting we intentionally wanted to give some guidance how to format this pretty in your Java source code and keep auto-formatters from breaking it.
The idea is to add every WHERE condition, every JOIN, every ORDER, GROUP_BY or HAVING to its own line with according indendation to keep it readable.